### ISSUE TITLE
Tab to close tag and move past

### ISSUE THREAD

**Issue by nianiam** [notaapp/nota#164]
In previous versions, when the cursor was within tags, the tab button could be used to 'move out of the tags and add a space'. I don't know if this was unintentionally removed, but this is no longer the behaviour of the editor. Now it does nothing, or indents the current paragraph.

If it was intended, I am requesting it be returned either by default or as a setting and indenting done another way (at the begging of the line, just after a bullet or at the the end of the line).

**Comment by astoilkov** [comment]
Yes, it was intended. We can reimplement the feature in the future but not set the default shortcut to Tab.

Here's why we removed it for now:
- It wasn't working properly. There were edge cases where it didn't do the right thing. For example, Tab didn't move from the link title to the link URL which, in our view, was the correct behavior.
- There were missing features that made the experience inconsistent. For example, you could move out of brackets but not out of square brackets, curly brackets, or quotes.
- Tab was too invasive of a shortcut. You could want to indent the line but you would move the caret instead. That was confusing for some users.

**Comment by nianiam** [comment]
This was one of my favourite features! 

I don't know if I'm an edge case here, but I use quite a lot of tags when documenting my code. When apps auto generate end tags (especially with two characters, \*\*) I find it painful having to either hit move several times or have to re-adjust my hands to option+right out of the tag. I appreciate that tab might not be the best key for this, but I still request the return of this feature in some form (preferably a toggle to allow tab to do what it does now, or move out of tags).

It seems like most of the inconsistent/missing features could be fixed using relatively simple regex matching. But I assume it's a bit more complicated than that!

**Comment by astoilkov** [comment]
Yes, I completely understand. We've actually used this feature ourselves.

We've actually started using it less when we discovered Emacs shortcuts and started using option+ctrl+f to move around (where our Caps Lock is remapped to Ctrl). We understand this isn't a good option for other users.

We're planning to reimplement the feature with the issues fixed and with support for all inline types. It will be a command that you will be able to assign Tab to in Preferences → Shortcuts.
